src/tests/Interop/COM/Activator/Program.cs

```diff
@@ -32,225 +32,308 @@ static void InvalidInterfaceRequest()
 
-            COMException e = Assert.Throws<COMException>(
-                () =>
-                {
-                    var CLSID_NotRegistered = new Guid("328FF83E-3F6C-4BE9-A742-752562032925"); // Random GUID
-                    var cxt = new ComActivationContext()
-                    {
-                        ClassId = CLSID_NotRegistered,
-                        InterfaceId = typeof(IClassFactory).GUID,
-                        AssemblyPath = @"C:\foo.dll"
-                    };
-                    ComActivator.GetClassFactoryForType(cxt);
-                },
-                "Class should not be found");
-
-            const int CLASS_E_CLASSNOTAVAILABLE = unchecked((int)0x80040111);
-            Assert.AreEqual(CLASS_E_CLASSNOTAVAILABLE, e.HResult, "Unexpected HRESULT");
-        }
-
-        static void ValidateAssemblyIsolation()
-        {
-            Console.WriteLine($"Running {nameof(ValidateAssemblyIsolation)}...");
-
-            string assemblySubPath = Path.Combine(Environment.CurrentDirectory, "Servers");
-            string assemblyAPath = Path.Combine(assemblySubPath, "AssemblyA.dll");
-            string assemblyBPath = Path.Combine(assemblySubPath, "AssemblyB.dll");
-            string assemblyCPath = Path.Combine(assemblySubPath, "AssemblyC.dll");
-            string assemblyPaths = $"{assemblyAPath}{Path.PathSeparator}{assemblyBPath}{Path.PathSeparator}{assemblyCPath}";
-
-            HostPolicyMock.Initialize(Environment.CurrentDirectory, null);
-
-            var CLSID_NotUsed = Guid.Empty; // During this phase of activation the GUID is not used.
-            Guid iid = typeof(IGetTypeFromC).GUID;
-            Type typeCFromAssemblyA;
-            Type typeCFromAssemblyB;
-
-            using (HostPolicyMock.Mock_corehost_resolve_component_dependencies(
-                0,
-                assemblyPaths,
-                string.Empty,
-                string.Empty))
+            if(!builtInComDisabled)
             {
-                var cxt = new ComActivationContext()
-                {
-                    ClassId = CLSID_NotUsed,
-                    InterfaceId = typeof(IClassFactory).GUID,
-                    AssemblyPath = assemblyAPath,
-                    AssemblyName = "AssemblyA",
-                    TypeName = "ClassFromA"
-                };
-
-                var factory = (IClassFactory)ComActivator.GetClassFactoryForType(cxt);
-
-                IntPtr svrRaw;
-                factory.CreateInstance(null, ref iid, out svrRaw);
-                var svr = (IGetTypeFromC)Marshal.GetObjectForIUnknown(svrRaw);
-                Marshal.Release(svrRaw);
-                typeCFromAssemblyA = (Type)svr.GetTypeFromC();
-            }
 
-            using (HostPolicyMock.Mock_corehost_resolve_component_dependencies(
-                0,
-                assemblyPaths,
-                string.Empty,
-                string.Empty))
+                COMException e = Assert.Throws<COMException>(
```

Comment:
This can be drastically simplified by not using the generic `Throws<T>`. Instead use `Throws()` and validate the type and details as expected.

Reply:
This is in reference to duplication in the branches, right?

Reply:
The actions that should throw are the same for built-in COM being disabled or not. Both here and in the test case above, you should be able to do one of:
- Define a variable for the action that should throw and pass that to `Assert.Throws`
- Use `Assert.Throws<Exception>` with `AssertThrowsOptions.AllowDerived` and validate the returned exception is of the type / has the properties expected based on whether or not built-in COM is disabled.